### PR TITLE
Fixing mailer and adding preview endpoint

### PR DESCRIPTION
**Pull request by dandehavilland**

### CODE REVIEW

File: app/views/notification_mailer/report_reschedule_email.text.erb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 An error occurred while attempting to process a completion:
 
-Attempt number <%=@attempt%> has been rescheduled for the next cycle.
+Attempt number <%=@attempts%> has been rescheduled for the next cycle.
 
 Vendor: <%=@integration.vendor_name %>
 Batch ID: <%=@task.batch_id %>
-Completion Type: <%=@action %>
```
Comment (betacar) [inline app/views/notification_mailer/report_reschedule_email.text.erb]: Same as above.

File: app/views/notification_mailer/report_failure_email.text.erb [hunk 2]
```diff
@@ -4,7 +4,6 @@ This task has either reached the maximum number of attempts, or is not suitable
 
 Vendor: <%=@integration.vendor_name %>
 Batch ID: <%=@task.batch_id %>
-Completion Type: <%=@action %>
```
[the remark this answers is not in the document]
Comment (dandehavilland) [reply app/views/notification_mailer/report_failure_email.text.erb]: I realised the `task` object (`Scheduler`) doesn't have knowledge of the action. The error message below will include detail on the reason for failure.

I can plumb that in if you think it's necessary.

[the remark this answers is not in the document]
Comment (betacar) [reply app/views/notification_mailer/report_failure_email.text.erb]: Yes, let's add it please.

**Pull request merged**